src/decoders/arw.rs

```diff
@@ -73,7 +73,7 @@ impl<'a> Decoder for ArwDecoder<'a> {
 }
 
 impl<'a> ArwDecoder<'a> {
-  fn image_a100(&self, camera: &Camera) -> Result<RawImage,String> {
```

Comment:
I wouldn't change the signature to these functions or to ok_camera. Passing the reference allows you to use it multiple times and sometimes that may be needed (e.g., first to decode, then to apply metadata).